lib/array.js

```diff
@@ -269,6 +308,21 @@ internals.fillMissedErrors = function (errors, requireds, state, options) {
     }
 };
 
+internals.fillOrderedErrors = function (errors, ordereds, state, options) {
+
+    for (var i = 0, il = ordereds.length; i < il; ++i) {
+        var presence = Hoek.reach(ordereds[i], '_flags.presence');
+        if (presence !== 'required') {
```

Comment:
There is no need to splice here, just build a new array out of the required schemas.